test/ruby/test_variable.rb

```diff
@@ -155,6 +155,19 @@ def test_include_refined_module_class_variable
     end
   end
 
+  def test_set_class_variable_on_frozen_object
+    set_cvar = EnvUtil.labeled_class("SetCVar") do
+      def self.set(val)
+        @@a = val # inline cache
+      end
+    end
```

Comment:
`class_eval` is needed.
```suggestion
    set_cvar = EnvUtil.labeled_class("SetCVar")
    set_cvar.class_eval "#{<<~"begin;"}\n#{<<~'end;'}"
    begin;
      def self.set(val)
        @@a = val # inline cache
      end
    end;
```

Reply:
So this pushes cref? Thanks!